hikari/internal/deprecation.py

```diff
@@ -35,72 +35,63 @@
+    additional_information: str
+        Additional information on the deprecation for the user.
 
     Other Parameters
     ----------------
-    version: typing.Optional[str]
-        If specified, the version it will be removed in.
-    alternative: typing.Optional[str]
-        If specified, the alternative to use.
     stack_level: int
         The stack level for the warning. Defaults to `3`.
     """
-    if inspect.isclass(obj) or inspect.isfunction(obj):
+    if inspect.isclass(obj):
+        action = ("Instantiation of", "class")
         obj = f"{obj.__module__}.{obj.__qualname__}"
+    else:
+        if inspect.isfunction(obj):
+            obj = f"{obj.__module__}.{obj.__qualname__}"
 
-    version_str = f"version {version}" if version is not None else "a following version"
-    message = f"'{obj}' is deprecated and will be removed in {version_str}."
+        action = ("Call to", "function/method")
 
-    if alternative is not None:
-        message += f" You can use '{alternative}' instead."
+    warnings.warn(
+        f"{action[0]} deprecated {action[1]} {obj!r} ({additional_information})",
+        category=DeprecationWarning,
+        stacklevel=stack_level,
+    )
 
-    warnings.warn(message, category=DeprecationWarning, stacklevel=stack_level)
 
+def deprecated(version: str, additional_information: str) -> typing.Callable[[T], T]:
+    """Mark a function or object as being deprecated.
 
-def deprecated(
-    version: typing.Optional[str] = None, alternative: typing.Optional[str] = None
-) -> typing.Callable[[T], T]:
-    """Mark a function as deprecated.
-
-    Other Parameters
-    ----------------
-    version: typing.Optional[str]
-        If specified, the version it will be removed in.
-    alternative: typing.Optional[str]
-        If specified, the alternative to use.
+    Parameters
+    ----------
+    version: typing.Any
+        The version this function or object is deprecated in.
+    additional_information: str
+        Additional information on the deprecation for the user.
     """
 
     def decorator(obj: T) -> T:
-        type_str = "class" if inspect.isclass(obj) else "function"
-        version_str = f"version {version}" if version is not None else "a following version"
-        alternative_str = f"You can use `{alternative}` instead." if alternative else ""
-
-        doc = inspect.getdoc(obj) or ""
-        doc += (
-            "\n"
-            "!!! warning\n"
-            f"    This {type_str} is deprecated and will be removed in {version_str}.\n"
-            f"    {alternative_str}\n"
-        )
-        obj.__doc__ = doc
+        old_doc = inspect.getdoc(obj)
+
+        # If the docstring is inherited we can assume that the deprecation warning was already added there
+        if old_doc:
+            first_line_end = old_doc.index("\n")
+            obj.__doc__ = (
+                old_doc[:first_line_end]
+                + f"\n\n.. deprecated:: {version}\n    {additional_information}"
```

Comment:
Wouldn't this lead to inconsistencies as to when the deprecation warnings are actually in the docs since it'll have to be done inline for cases where specific arguments were deprecated?

Reply:
This is aimed for fully deprecated functions/methods/classes. When deprecating only an argument we can add the warning to the docstring and then raise the warning with `warn_deprecated`

Reply:
Which makes the docs inconsistent for anybody reading through the code right?

Reply:
Not really, since the decorator would be there. The deprecation decorator could also be extended to include arguments and detect when a specific argument is passed